Doesn't work for me.
```
building path(s) ‘/nix/store/4izhldawvi464zykg3sl3adj50czj182-vimplugin-youcompleteme-2015-05-29’
unpacking sources
unpacking source archive /nix/store/dbjmi2f897cvqaign3h7h08q3mvr5mcd-youcompleteme-c2d962d
source root is youcompleteme-c2d962d
patching sources
configuring
building
patching script interpreter paths in .
./install.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/bash "
./print_todos.sh: interpreter directive changed from "/bin/bash" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/bash  "
./run_tests.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/bash "
./third_party/pythonfutures/setup.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/requests/requests/packages/chardet/chardetect.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/requests/setup.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/requests/test_requests.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/build.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/scripts/fuse_gtest_files.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/scripts/gen_gtest_pred_impl.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/scripts/gtest-config.in: interpreter directive changed from "/bin/sh" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/sh  "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/scripts/pump.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/scripts/release_docs.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/scripts/upload.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/scripts/upload_gtest.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_break_on_failure_unittest.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_catch_exceptions_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_color_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_env_var_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_filter_unittest.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_help_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_list_tests_unittest.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_output_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_shuffle_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_test_utils.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_throw_on_failure_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_uninitialized_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_xml_outfiles_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_xml_output_unittest.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_xml_test_utils.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/fuse_gmock_files.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/generator/cpp/ast.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/generator/cpp/gmock_class.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/generator/cpp/gmock_class_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/generator/cpp/keywords.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/generator/cpp/tokenize.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/generator/cpp/utils.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/generator/gmock_gen.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/gmock-config.in: interpreter directive changed from "/bin/sh" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/sh  "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/gmock_doctor.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/upload.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/upload_gmock.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/test/gmock_leak_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/test/gmock_output_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/test/gmock_test_utils.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/examples/example_client.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/run_tests.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/bash "
./third_party/ycmd/style_format.sh: interpreter directive changed from "/bin/bash" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/bash  "
./third_party/ycmd/third_party/bottle/docs/_locale/update.sh: interpreter directive changed from "/bin/bash" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/bash  "
./third_party/ycmd/third_party/bottle/setup.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/third_party/bottle/test/testall.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/third_party/bottle/test/travis_setup.sh: interpreter directive changed from "/bin/bash" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/bash  "
./third_party/ycmd/third_party/gocode/_testing/all.bash: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/bash "
./third_party/ycmd/third_party/gocode/_testing/run.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/third_party/gocode/debian/rules: interpreter directive changed from "/usr/bin/make -f" to "/nix/store/hv8pgyn6a0a5h9dyd7s7cwh5105gykj5-gnumake-4.1/bin/make -f "
./third_party/ycmd/third_party/gocode/vim/pathogen_update.sh: interpreter directive changed from "/bin/sh" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/sh  "
./third_party/ycmd/third_party/gocode/vim/symlink.sh: interpreter directive changed from "/bin/sh" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/sh  "
./third_party/ycmd/third_party/gocode/vim/update.sh: interpreter directive changed from "/bin/sh" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/sh  "
./third_party/ycmd/third_party/jedi/scripts/memory_check.py: interpreter directive changed from " /usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/third_party/jedi/scripts/profile.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/third_party/jedi/scripts/wx_check.py: interpreter directive changed from " /usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/third_party/jedi/setup.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/third_party/jedi/sith.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/third_party/jedi/test/refactor.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/third_party/jedi/test/run.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/third_party/requests/requests/packages/chardet/chardetect.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/third_party/requests/setup.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/third_party/requests/test_requests.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/update_boost.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/bash "
./third_party/ycmd/ycmd/__main__.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/ycmd/completers/all/identifier_completer.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/ycmd/completers/cpp/clang_completer.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/ycmd/completers/cs/cs_completer.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
./third_party/ycmd/ycmd/completers/go/gocode_completer.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python "
-- The C compiler identification is GNU 4.8.4
-- The CXX compiler identification is GNU 4.8.4
-- Check for working C compiler: /nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/gcc
-- Check for working C compiler: /nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/g++
-- Check for working CXX compiler: /nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Your C++ compiler supports C++11, compiling in that mode.
-- Found PythonLibs: /nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/lib/libpython2.7.so (found suitable version "2.7.10", minimum required is "2.6") 
-- Found PythonInterp: /nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python2 (found suitable version "2.7.10", minimum required is "2.6") 
Using libclang to provide semantic completion for C/C++/ObjC
Using external libclang: TEMP-NOTFOUND
-- Found PythonInterp: /nix/store/z6vp5aix4ks1zdjdry7v7dahg8dd02sy-python-2.7.10/bin/python2 (found version "2.7.10") 
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
TEMP
    linked by target "ycm_core" in directory /nix/store/4izhldawvi464zykg3sl3adj50czj182-vimplugin-youcompleteme-2015-05-29/share/vim-plugins/youcompleteme/third_party/ycmd/cpp/ycm

-- Configuring incomplete, errors occurred!
See also "/nix/store/4izhldawvi464zykg3sl3adj50czj182-vimplugin-youcompleteme-2015-05-29/share/vim-plugins/youcompleteme/build/CMakeFiles/CMakeOutput.log".
See also "/nix/store/4izhldawvi464zykg3sl3adj50czj182-vimplugin-youcompleteme-2015-05-29/share/vim-plugins/youcompleteme/build/CMakeFiles/CMakeError.log".
builder for ‘/nix/store/hvzrwk5hyyq9ghyixrma672r1nqwfa8h-vimplugin-youcompleteme-2015-05-29.drv’ failed with exit code 1
```

--- a/pkgs/misc/vim-plugins/default.nix
+++ b/pkgs/misc/vim-plugins/default.nix
@@ -685,7 +685,8 @@ rec {
     dependencies = [];
     buildInputs = [
       python cmake
-      llvmPackages.clang-unwrapped llvmPackages.llvm
+      (if stdenv.isDarwin then llvmPackages.clang else llvmPackages.clang-unwrapped)
+      llvmPackages.llvm
     ];
 
     configurePhase = ":";

